protocol/driver.go

```diff
@@ -190,3 +228,108 @@ func (m *ParseResponse) Nodes() (nodes.Node, error) {
 	}
 	return ast, err
 }
+
+// Version implements DriverHostClient.
+func (c *client) Version(rctx context.Context) (driver.Version, error) {
+	sp, ctx := opentracing.StartSpanFromContext(rctx, "bblfsh.client.Version")
+	defer sp.Finish()
+
+	resp, err := c.h.ServerVersion(ctx, &VersionRequest{})
+	if err != nil {
+		return driver.Version{}, err
+	}
+	vers := (*driver.Version)(resp.Version)
+	return *vers, nil
+}
+
+// Languages implements DriverHostClient.
+func (c *client) Languages(rctx context.Context) ([]manifest.Manifest, error) {
+	sp, ctx := opentracing.StartSpanFromContext(rctx, "bblfsh.client.Languages")
+	defer sp.Finish()
+
+	resp, err := c.h.SupportedLanguages(ctx, &SupportedLanguagesRequest{})
+	if err != nil {
+		return nil, err
+	}
+	out := make([]manifest.Manifest, 0, len(resp.Languages))
+	for _, m := range resp.Languages {
+		dm := m.ToNative()
+		out = append(out, *dm)
+	}
+	return out, nil
+}
+
+// NewManifest converts driver manifest to the corresponding protocol message.
+func NewManifest(m *manifest.Manifest) *Manifest {
+	dm := &Manifest{
+		Name:     m.Name,
+		Language: m.Language,
+		Aliases:  m.Aliases,
+		Features: make([]string, 0, len(m.Features)),
+	}
+	if m.Version != "" || !m.Build.IsZero() {
+		dm.Version = &Version{
+			Version: m.Version,
+			Build:   m.Build,
+		}
+	}
+	switch m.Status {
+	case manifest.Inactive:
+		dm.Status = DevelopmentStatus_Inactive
+	case manifest.Planning:
+		dm.Status = DevelopmentStatus_Planning
+	case manifest.PreAlpha:
+		dm.Status = DevelopmentStatus_PreAlpha
+	case manifest.Alpha:
+		dm.Status = DevelopmentStatus_Alpha
+	case manifest.Beta:
+		dm.Status = DevelopmentStatus_Beta
+	case manifest.Stable:
+		dm.Status = DevelopmentStatus_Stable
+	case manifest.Mature:
+		dm.Status = DevelopmentStatus_Mature
+	default:
+		st, _ := strconv.Atoi(string(m.Status))
+		dm.Status = DevelopmentStatus(st)
+	}
+	for _, f := range m.Features {
+		dm.Features = append(dm.Features, strings.ToLower(string(f)))
+	}
+	return dm
+}
+
+// ToNative converts the manifest message to the driver manifest used by the SDK.
+func (m *Manifest) ToNative() *manifest.Manifest {
+	dm := &manifest.Manifest{
+		Name:     m.Name,
+		Language: m.Language,
+		Aliases:  m.Aliases,
+		Features: make([]manifest.Feature, 0, len(m.Features)),
+	}
+	if m.Version != nil {
+		dm.Version = m.Version.Version
+		dm.Build = m.Version.Build
+	}
+	switch m.Status {
```

Comment:
The generated code includes a map for these values, and I was going to suggest we use it, but "pre-alpha" differs even folding case. 😢

Reply:
Yep :( Although we can make a rule that upper case in the middle of the word converts to a `-` in the lower case. All the values seem to follow this rule.